controllers/machinehealthcheck_targets.go

```diff
@@ -44,6 +44,11 @@ const (
 	EventDetectedUnhealthy string = "DetectedUnhealthy"
 )
 
+var (
+	// We allow users to disable the nodeStartupTimeout by setting the duration to 0.
+	disabledNodeStartupTimeout = metav1.Duration{Duration: 0}
```

Comment:
We could call it ZeroDuration, or EmptyDuration and export it in a different file

Reply:
Yeah I can go ahead with that. @vincepri Since you suggested the `IsZeroDuration` kind of method before, do you have a preference for which way we go in terms of comparing to a public var vs having a helper?

Reply:
Second question, if we just were to compare against a `ZeroDuration`, does that make the code harder to read, would it not be preferable to have a local alias still so that the if statement is obvious

I'm thinking
```
if timeoutForMachineToHaveNode.Seconds() == disabledNodeStartupTimeout.Seconds() {
```
vs
```
if timeoutForMachineToHaveNode.Seconds() == clusterv1.ZeroDuration.Seconds() {
```

Reply:
the local alias version does make it extremely clear what is happening

Reply:
Given that Durations are usually non pointers, we don't have to use the `Seconds()` here, a plain == should work well

Reply:
Have pushed a commit that adds this constant in the API, LMK what you think